perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1543,55 +1546,68 @@ sub process_confirmation : Private {
-        $problem->update({ lastupdate => \'current_timestamp' });
+        foreach my $problem (@problems) {
+            $problem->confirm;
+            $problem->update({ lastupdate => \'current_timestamp' });
 
-        # Subscribe problem reporter to email updates
-        $c->forward( '/report/new/create_related_things', [ $problem ] );
-    }
-
-    # log the problem creation user in to the site
-    if ( $data->{name} || $data->{password} ) {
-        if (!$problem->user->email_verified) {
-            $problem->user->email( $data->{email} ) if $data->{email};
-        } elsif (!$problem->user->phone_verified) {
-            $problem->user->phone( $data->{phone} ) if $data->{phone};
+            # Subscribe problem reporter to email updates
+            $c->forward( '/report/new/create_related_things', [ $problem ] );
         }
-        $problem->user->password( $data->{password}, 1 ) if $data->{password};
-        for (qw(name title facebook_id twitter_id)) {
-            $problem->user->$_( $data->{$_} ) if $data->{$_};
+
+        if ($token_redeem_cooldown_seconds) {
+            # Mark when the token was redeemed if there is a cooldown set so
+            # we can track if new uses are within the window.
+            $token->mark_redeemed(time());
         }
-        $problem->user->add_oidc_id($data->{oidc_id}) if $data->{oidc_id};
-        $problem->user->extra({
-            %{ $problem->user->get_extra() },
-            %{ $data->{extra} }
-        }) if $data->{extra};
-
-        $problem->user->update;
-
-        # Make sure extra oauth state is restored, if applicable
-        foreach (qw/logout_redirect_uri change_password_uri/) {
-            if ($data->{$_}) {
-                $c->session->{oauth} ||= ();
-                $c->session->{oauth}{$_} = $data->{$_};
+
+        # update user details
+        if ( $data->{name} || $data->{password} ) {
+            if (!$problem->user->email_verified) {
+                $problem->user->email( $data->{email} ) if $data->{email};
+            } elsif (!$problem->user->phone_verified) {
+                $problem->user->phone( $data->{phone} ) if $data->{phone};
+            }
+            $problem->user->password( $data->{password}, 1 ) if $data->{password};
+            for (qw(name title facebook_id twitter_id)) {
+                $problem->user->$_( $data->{$_} ) if $data->{$_};
+            }
+            $problem->user->add_oidc_id($data->{oidc_id}) if $data->{oidc_id};
+            $problem->user->extra({
+                %{ $problem->user->get_extra() },
+                %{ $data->{extra} }
+            }) if $data->{extra};
+            $problem->user->update;
+
+            # Make sure extra oauth state is restored, if applicable
+            foreach (qw/logout_redirect_uri change_password_uri/) {
+                if ($data->{$_}) {
+                    $c->session->{oauth} ||= ();
+                    $c->session->{oauth}{$_} = $data->{$_};
+                }
```

Comment:
Sorry, managed to let the fixup get squashed on a rebase, but only change is this block moving back in to the above if block.